Languages/en_US/Themes.php

```diff
@@ -165,4 +167,14 @@
 $txt['og_image'] = 'Open Graph image';
 $txt['og_image_desc'] = 'Suggested size: 175x175px. <a href="https://ogp.me/" target="_blank" class="bbc_link">Open Graph</a> is used for social media sharing.';
 
+
+// Theme Mode (dark, light, system, etc)
+$txt['theme_pick_colormode'] = 'Select Color Mode';
+$txt['theme_colormode_default'] = 'Default color mode';
+$txt['theme_colormode_user_disable'] = 'Disable user color mode selection';
+$txt['colormode_light'] = 'Light';
+$txt['colormode_dark'] = 'Dark';
+$txt['colormode_system'] = 'System Default';
```

Comment:
I think these should go into ThemeStrings because they are theme-specific.

Reply:
...or not  ... this commit only supports the three that the browser supports and is hardcoded intothe souurces.

Reply:
It would now be part of the core, it's what it will supports by default for any theme. Sleepy said we could have more things like a contrast mode or similar so it can just be added there with a different file. I think using the data attribute is also just our best option to do this, seems at least Xenforo does something similar.

Reply:
If we want a high contrast mode for third party themes, shouldn't the list of modes be defined in template_init?

The data attributes are a nice addition

Reply:
My assumption is that it would be a limited list of things, we just haven’t decided their names and distinction.
Contrast mode seems to be the same as “night” mode to me? It just needs to be discussed a bit more.

Reply:
I'm thinking these scenarios (considering just contrast mode):
- Someone only wants a contrast mode.
- Someone wants both (and remove color in contrast.css)

How do we deal with the options available, a new array?
User doesn't want to see "Contrast" as an option if it's not included/taken care of.

I suppose we remove _has_dark_mode_ and instead use:
color_modes = ['dark', 'contrast'];

And add them to the array since we expect both "Light" and "System" by default when we have color modes.

Only restriction is that the conditional for "dark" needs to be exist because it will load differently, to include the media query in the attributes.

What do you think?

Reply:
> I suppose we remove _has_dark_mode_ and instead use:
> color_modes = ['dark', 'contrast'];

Yeah, probably best this way. What if a theme is only light or only dark? Should this array be    `['system', 'light', 'dark']` to give theme more control?

Reply:
You're now overthinking it, that was my mistake too when I built my themes dark mode 😆
"Only light" or "only dark" means the same! You can't have dark mode without both and you can always define the default in the admin.

In fact, we might want to backpedal on this array and keep it bool.
I haven't really found much info on color modes, we just know of the existence of a "contrast mode". But is it really a color mode or would it be a variant? It's technically just black and white.

Reply:
b&w sounds like a plain olds fasshioned variant